### PR TITLE
Chore/improve service error

### PR DESCRIPTION
**Pull request by eh-am**
Closes https://github.com/pyroscope-io/pyroscope/issues/941

# Displays more info to the user
image

We can assume most users of pyroscope are quite technical and would like to know as much as possible when something fails. Hence we display the `error.message` in a notification message

# Log full errors to the console
In the same manner, we also log the entire Error object to the console (using a redux middleware):
image

In the future we can report these kind of errors to the server, so that we can understand better what kind of errors our users have.

### CODE REVIEW

File: webapp/javascript/redux/store.ts [hunk 1]
```diff
@@ -41,9 +42,18 @@ const reducer = combineReducers({
...
-    // TODO: it would be nice to have an actual error here
-    //    console.error(action.error.stack);
+    // since redux-toolkit serializes errors
+    // we should deserialize them back
+    // https://github.com/reduxjs/redux-toolkit/blob/db0d7dc20939b62f8c59631cc030575b78642296/packages/toolkit/src/createAsyncThunk.ts#L94
+    try {
+      const deserialized = deserializeError(action.error);
+      console.error(deserialized);
+
+      // TODO: report error to server?
```
Comment (eh-am) [inline webapp/javascript/redux/store.ts:52]: @petethepig here's where the stats api would be useful

File: webapp/javascript/services/base.ts [hunk 2]
```diff
@@ -107,23 +117,35 @@ export async function request(
         }
       }
```
Comment (eh-am) [inline webapp/javascript/services/base.ts:119]: @abeaumont @kolesnikovae
you guys may know this better, is any other case we are missing?
eg server responds with an error message in a different field other than errors | error | message

File: webapp/javascript/redux/store.ts (hunk 1 above)
Comment (petethepig) [reply webapp/javascript/redux/store.ts:52]: agreed

File: webapp/javascript/services/base.ts (hunk 2 above)
Comment (abeaumont) [reply webapp/javascript/services/base.ts:119]: I think we are not sending errors in JSON format in all the cases. Some consistency in the API would be helpful here.

**Pull request merged**